runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4788,341 +4800,332 @@ J9::Z::TreeEvaluator::anewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg
+   // If number of leaf arrays is zero, jump to the end.
+   TR::LabelSymbol *controlFlowEndLabel = generateLabelSymbol(cg);
+   controlFlowEndLabel->setEndInternalControlFlow();
+   cursor = generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CL, node, scratchReg, 0, TR::InstOpCode::COND_BE, controlFlowEndLabel, false /* needsCC */, cursor);
+
+   cursor = generateRXInstruction(cg, TR::InstOpCode::LA, node, dim1SizeReg, generateS390MemoryReference(dim1SizeReg, resultReg, 0, cg), cursor);
+   iComment("dim1SizeReg points to the fist leaf array.");
+   // Load component class to class register. In case of comp refs, the calss is 32bits long so using this register for leaf array length value as well.
+   // ClassReg when non-comp refs: |     class      |
+   // ClassReg when comp refs:     | class | length |
+   cursor = generateRXInstruction(cg, TR::InstOpCode::LG, node, classReg, generateS390MemoryReference(classReg, offsetof(J9ArrayClass, componentType) + (useCompRefs ? 4 : 0), cg), cursor);
+   int32_t shiftAmount = TR::Compiler->om.compressedReferenceShift();
+   if (useCompRefs)
+      {
+      // Use the class register for both class and length fields.
+      cursor = generateRXInstruction(cg, TR::InstOpCode::L, node, classReg, generateS390MemoryReference(dimsPtrReg, 0, cg), cursor);
+      if (shiftAmount > 0)
+         {
+         // Keep a compressed dim2 size in higher 32 bits.
+         // dim2SizeReg when shift amount is 0:    |          unused            |    leaf array size    |
+         // dim2SizeReg when shift amount is > 0:  | compressed leaf array size |    leaf array size    |
+         cursor = generateRIEInstruction(cg, TR::InstOpCode::RISBG, node, dim2SizeReg, dim2SizeReg, 0, 31, 32-shiftAmount, cursor);
+         iComment("Compressed dim2 size in higher 32bits.");
+         // Keep a compressed reference in higher 32bits of scratchReg.
+         // scratchReg when shift amount is 0:    |       leaf array length       |    number of leaf arrays    |
+         // scratchReg when shift amount is > 0:  | compressed leaf array address |    number of leaf arrays    |
+         cursor = generateRIEInstruction(cg, TR::InstOpCode::RISBG, node, scratchReg, dim1SizeReg, 0, 31, 32-shiftAmount, cursor);
+         iComment("Compressed dim2 reference in higher 32 bits.");
+         }
+      }
+
+   // Load the address of the first element of the first dimension array in sizeReg.
+   cursor = generateRXInstruction(cg, TR::InstOpCode::LA, node, sizeReg,
+      generateS390MemoryReference(resultReg, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg), cursor);
+   // Start setting second dim:
+   TR::LabelSymbol *secondDimLabel = generateLabelSymbol(cg);
+   cursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, secondDimLabel, cursor);
+   // Store the class field. In case of comp refs, store the class and length fields.
```

Comment:
Please update the short for `comp refs` to `compressedrefs` throughout this evaluator.